rewrite-gradle/src/main/java/org/openrewrite/gradle/UpgradeDependencyVersion.java

```diff
@@ -116,189 +120,295 @@ public Validated<Object> validate() {
+        MethodMatcher dependencyDsl = new MethodMatcher("DependencyHandlerSpec *(..)");
+        return new GroovyVisitor<ExecutionContext>() {
             @Override
-            public J postVisit(J tree, ExecutionContext ctx) {
-                if (tree instanceof JavaSourceFile) {
-                    JavaSourceFile cu = (JavaSourceFile) tree;
-                    Map<String, Map<GroupArtifact, Set<String>>>  variableNames = getCursor().getMessage(VERSION_VARIABLE_KEY);
-                    if (variableNames != null) {
-                        Optional<GradleProject> maybeGp = cu.getMarkers()
-                                .findFirst(GradleProject.class);
-                        if (!maybeGp.isPresent()) {
-                            return cu;
-                        }
-
-                        cu = (JavaSourceFile) new UpdateVariable(variableNames, maybeGp.get()).visitNonNull(cu, ctx);
-                    }
-                    Map<GroupArtifactVersion, Set<String>> versionUpdates = getCursor().getMessage(NEW_VERSION_KEY);
-                    if (versionUpdates != null) {
-                        Optional<GradleProject> maybeGp = cu.getMarkers()
-                                .findFirst(GradleProject.class);
-                        if (!maybeGp.isPresent()) {
-                            return cu;
+            public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                J.MethodInvocation m = (J.MethodInvocation) super.visitMethodInvocation(method, ctx);
+                if (dependencyDsl.matches(m)) {
+                    if(m.getArguments().get(0) instanceof G.MapEntry) {
+                        String groupId = null;
+                        String artifactId = null;
+                        String version = null;
+
+                        for (Expression e : m.getArguments()) {
+                            if (!(e instanceof G.MapEntry)) {
+                                continue;
+                            }
+                            G.MapEntry arg = (G.MapEntry) e;
+                            if (!(arg.getKey() instanceof J.Literal)) {
+                                continue;
+                            }
+                            J.Literal key = (J.Literal) arg.getKey();
+                            String valueValue = null;
+                            if(arg.getValue() instanceof J.Literal) {
+                                J.Literal value = (J.Literal) arg.getValue();
+                                if(value.getValue() instanceof String) {
+                                    valueValue = (String)value.getValue();
+                                }
+                            } else if(arg.getValue() instanceof J.Identifier) {
+                                J.Identifier value = (J.Identifier) arg.getValue();
+                                valueValue = value.getSimpleName();
+                            }
```

Comment:
If you're up for it, we should also go ahead and attempt to cover the `G.GString` case coming in here for the version. Much like what was described in on of the PR comments:

```
implementation group: "com.google.guava", name: "guava", version: "${guavaVersion}"
```